### DISCUSSION TITLE
03. Can an open source solution be trusted to be accurate? What are the relevant considerations?

### DISCUSSION THREAD

**Discussion by MichaelRimler** [phuse-org/OSTCDA#3] [General]
Can an open source solution be trusted to be accurate? What are the relevant considerations?

**Comment by MichaelRimler** [comment]
Generally, yes - but it falls on the user (sponsor) to ensure their trust in it. Tend to use QC and QA processes to assess the human actions taken in our data handling - and testing/validation for the machine actions. Open source available solutions take a human action and ask a machine to do it. Therefore, packages with robust internal testing - good documentation - and a good usage history are more likely to be trusted over those with less testing, documentation, and usage. 

Testing in R: https://github.com/r-lib/testthat 

Assessing risk in R packages (establishing a level of trust): https://github.com/pharmaR/riskmetric

PHUSE Blog
https://phuse.global/Communications/PHUSE_Blog/can-clinical-data-processed-with-r-be-used-in-a-regulatory-submission

**Comment by rossfarrugia** [comment]
Probably best to reference R Validation Hub here and their white paper: https://www.pharmar.org/

**Comment by erictmsun** [comment]
If stress a bit more on Trustfulness than the Accuracy mentioned above, the SBOM (URL below) shall be introduced for the integrity and security of the Open Sources, either at element or package levels, to be used in highly regulated data analysis and reporting in the life science / pharma RnD domains.
https://www.ntia.gov/page/software-bill-materials#:~:text=A%20%E2%80%9CSoftware%20Bill%20of%20Materials,that%20make%20up%20software%20components.

Also, not for a promotion but info sharing, tool from checkmarx scans all source codes, open, proprietary or mixed, for better security, code preservation and even more for version control; it depends on how critical the project / tool is especially if will face the GxP auditing.
